### PR TITLE
openai, update tsp for *AsPlainText op, request as multipart/form-data, and response as text/plain

### LINKED ISSUE #26299: [Azure Open AI] getAudioTranscriptionAsPlainText operation does not produce content-type multipart/form-data
`getAudioTranscriptionAsPlainText` operation at
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/OpenAI.Inference/routes/audio_transcription.tsp#L29-L34

From [playground](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAYXp1cmUtdG9vbHMvdHlwZXNwZWMtxhVjb3JlIjsKySvIHy9yZXN01RlodHRw1Rl2ZXJzaW9uaW5nIjsKCnVzaW5nIFR5cGVTcGVjLlJlc3Q70BVIdHRw0RVWyUfIG0HkAKouQ29yZS5UcmFpdHPSGTsKCkBzZXJ2aWNlKHt0aXRsZTogIlRleHQgYXV0aG9y5ACVfSkKQOcApGVkKFPGLsducykKbmFtZXNwYWNlx1dPcGVuQUk7CgplbnVtIM8vIHsKICBAdXNlRGVwZW5kZW5jeSjrAKfIJy52MV8wX1ByZXZpZXdfMikKICB2MjAyM18wOV8wMcgZOiAixBUtMDktMDEtcMYVIiwKfQoKaW50ZXJm5ACVUmVzb3VyY2VPcGVyYXTEXQogIGV4dGVuZHPsASLSKDxOb1JlcGVhdGFibGVSZXF1ZXN0cyAmIE5vQ29uZGnEJGFszhhsaWVudMcTSWQ%2BIHt9CgovLyBUaGUgb3JkZXIgaW4gd2hpY2ggdMQTyWsgYXJlIGRlZmluZWQgbWF0dGVycyBmb3LFJ3B1cnBvc2Ugb2Yg5QFcUEl2MiBnZW7HPQoKQGRvYygiQSDkAk5pZmljIGRlcGxveW1lbnQiKQpA7QJMLnLnAOgoIsomcyIpCm1vZGVsIETJFOUBqiAgQHZpc2liaWxpdHkoInJlYWQi5AGNICDGdMRXaWZpZXMgZWl0aGXmALDGSspeIOQCIiAod2hlbiDrAoDmAM1JKSBvcscz0Shub24tzix0byB1c2XnARtpcyBy5gFsLugAlnByb2plY3RlZE5hbWUoImphdmEiLCDrAOtPck3EbMQfyDRrZXnFCcolSWQ6IHN05AL9O%2BQCTeYA7CIiCuQByGNvbmZpZ3XmAXwgaW5mb3JtxgzkAJVhbiBhdWRpbyB0cmFuc2NyaXDFG%2BgApwoiIukBa0HEKFTMJ0%2FFBucDJMh0IOUCPsZVZGF0YeQBA8hdYmUuIFTkAQdtdXN0IGJl5QF%2FYmluYXJ55ACodOQBfW9mIGEgZmlsZeQCfG9u5QJHxCdzdXDkBInkAnBlZGlh5AC%2BbWF0czrkAQtmbGFjLCBtcDPEBTTEBWVnxAZnYSwgbTRhLCBvZ2csIHdhdiwgd2VibS4KICDlANnyAYFjc2hhcnDkAYPlAO9EYXRhxSjkAJE6IGJ5dGVzOwryAO9wcmlt5ADFc3Bva2VuIGxhbmd1YWfpALzuAQ7kAPjpARFkLOUA2mxpZWQgYXMgYSB0d28tbGXkA1IgSVNPLTYzOS0xylFjb2RlCiAgc3VjaMQvJ2VuJ%2BQCkCdmcifkANZQcm92aWTkAovlAm5rbm935AH0cHV0ykNpcyBv5QG5YWwgYnV0IG1heSBpbXByb3bmAZBhY2N1cmFjeSBhbmQvb3IgbGF05AT35QDM7AH%2B6wFGyFw%2F6gKH7gEkQW7KdGhpbnTkAQdndWlkxnLlAmAncyBzdHls5AR55QIMaW51ZSBmcm9tIGHkAV%2FkAqzlAUZzZWfkAvPkAkllIHdyaXR08gFxCiAgcHJvbXB0IHNob3Vs5QSj5wTE%2FwGq6gGq6wDfxkz5AN3kAIlzYW1wbOUBeGVt5QUmdXJlLCBiZXR3ZWVuIDDkAUwgMcRTSGln5AR4dmFsdWVzIGxpa2UgMC44IHdpbGwgbWFr5gERb3V05AGzbW9yZSByYW5kb20s5AWHbGUgbG930T4yyz5pxzZmb2N1c%2BQCU25kIGRldGVybWluaXN0aWPkAIRJZiBzZeUBhDAs6wUCxUbkBK5sb2fkAPRiYeYFSMQpYXV0b%2BQEN2NhbGx5IGluY3JlYXPmAK3rAPIgdW50aWwgY2VydGFpbiB0aHJlc2hvbGTmBjBoaXTrAVXLNz86IGZsb2F0MzL0AVvmAJ3wBUn3BLDJXy8vIEltcGxl5AIs5gTvbm90ZTogZGV2ZWzkBsktZmFj5AGx6AaFxiVvZuwGCekF6mJ5IGPlByxz6AJSYuUFWnRyb2xsZWTnBk7Gd3ZpYeQFV%2BkHLSBwYXJhbeQBhMRQYnntBgkgYm9keSBmaWVsZCAtLeUDkG9ubHnqBPlvc%2BgFNMYlyn9oaWRkZW4gYuQGDS8v6QCgaWbUe%2BYBgukCE3BvcHXkA9hkIGlu5AWgaGXuAJFvbuUGGnMtbmVlZGVkIGJhc2lz5AFe%2BgUwSeQIwm5hbE5vbuUHA%2B4Gt2DlAUhg6wNE6QaoUmVzdWx09Aaa6gDIdGhhdOwFNegDv%2BYDr%2BUAxOgEC3RleOUHX%2FgGtesGruQCXsxVxDvqCMblBUvkATDqBBPnArl4dO0Ax2FsaWFzIE11bHRpcGFydEZvcm3kBjjnCVdIZWFkZXJz5grRID3yCuUu0yk85wjN6QCn6Ace5AuF6QCj6QEgLiBBbHdheeQHWOcAhS%2FkAU0t5QZyxi7lBfbIL%2BgISmjFcygix14txF7HHMcT5AmMOiAi01giOwp9PuQLj%2BgDrkdldOoDl%2BkBT%2BQCbmFzc29jaeUCb21ldGHEPuYF2ukBZO0Bz8QgLuYBuuYElO4HMeQIGOQF5PEF%2FGNvcnJlc3DkCuzkBYnmAtfJHml0IHdhc8hqaW7nCNxAYWPkARNTZeQDK3RvcigiL8oWKCLlAJIv7QRt5Ap9QOQC92VkUm91dGUKb3AgZ2V08gJqQXNQbGFpbuUMnmntC7xGb3VuZOYLSvILyDzlAYDqCtksxRDSX%2BcJfizlA%2F7mAPlzZV%2FmA0XpCWHmBEg65QGMLCBzcnQsIHZ0dMVS5gLCxl7%2FAr3mAqkKPjsK&e=%40azure-tools%2Ftypespec-autorest&options=%7B%7D), it does not produce content-type "multipart/form-data".

I assume the request of this operation is content-type "multipart/form-data". If not, just close the issue.

This causes SDK issue as well. The request of this will not be as `content-type: multipart/form-data`.

The `MultipartFormDataRequestHeadersTraits` seems not work with `Azure.Core.Foundations.ResourceOperation` template.

I can write
```ts
op getAudioTranscriptionAsPlainText is Azure.Core.Foundations.ResourceOperation<
  Deployment,
  AudioTranscriptionOptions & Azure.Core.Traits.Private.TraitProperties<MultipartFormDataRequestHeadersTraits, TraitLocation.Parameters, TraitContext.Action>,
  string,
  MultipartFormDataRequestHeadersTraits
>;
```
to make it work as "multipart/form-data". But I don't know whether this is good TypeSpec or not.

@jpalvarezl @mssfang

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
fix https://github.com/Azure/azure-rest-api-specs/issues/26299

## Reason for the change:

`Azure.Core.Foundations.ResourceOperation` template not support the `MultipartFormDataRequestHeadersTraits` (service should avoid `Azure.Core.Foundations` in the first place)
Changed to use `Azure.Core.ResourceAction`.

## Reason for the breaking changes in Swagger

No change on backend. The only change is the TypeSpec that describes the backend, and the generated Swagger.

It had been an oversight when writing the typespec source. The Trait is defined to operation but not have effect with `Azure.Core.Foundations.ResourceOperation`
There is an ongoing bug on typespec-azure-core https://github.com/microsoft/typespec/issues/2938

The `GetAudioTranscriptionAsResponseObject`/`GetAudioTranscriptionAsPlainText` REST API is a same API, and is always a [multipart/form-data request](https://github.com/Azure/azure-rest-api-specs/blob/91a268c2e1ebead1dea537137c7008e8d019d886/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-09-01-preview/generated.json#L502-L541).

Request model is [AudioTranscriptionOptions](https://github.com/Azure/azure-rest-api-specs/blob/91a268c2e1ebead1dea537137c7008e8d019d886/specification/cognitiveservices/OpenAI.Inference/models/audio/audio_transcription.tsp#L8-L85). When `responseFormat=json`, it be `GetAudioTranscriptionAsResponseObject` and has response content-type=application/json. When `responseFormat=text`, it be `GetAudioTranscriptionAsPlainText` and has response content-type=text/plain.

Therefore, this PR is a "fix" on the incorrect Swagger, not introducing breaking changes.

# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/routes/audio_transcription.tsp [hunk 1]
```diff
@@ -26,11 +26,14 @@ written language corresponding to the language it was spoken in.
...
+op getAudioTranscriptionAsPlainText is Azure.Core.ResourceAction<
   Deployment,
   AudioTranscriptionOptions,  // response_format must be one of: text, srt, vtt
-  string,
-  MultipartFormDataRequestHeadersTraits
+  {
+    @body
+    body: string
+  },
+  MultipartFormDataRequestHeadersTraits & TextPlainResponseHeadersTraits
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/OpenAI.Inference/routes/audio_transcription.tsp:36]: I've put the content-type of response to traits as well.

File: specification/cognitiveservices/OpenAI.Inference/models/audio/common.tsp [hunk 2]
```diff
@@ -13,3 +15,9 @@ enum AudioTaskLabel {
   @doc("Accompanying response data resulted from an audio translation task.")
   translate: "translate",
 }
+
+alias AudioTextResponse = {
+  @doc("The text response data.")
```
Comment (jpalvarezl) [inline specification/cognitiveservices/OpenAI.Inference/models/audio/common.tsp:20]: I would phrase this like we did for the request header trait:

@doc("The content type for the response. Always text/plain for this operation.")

**Review by jpalvarezl — APPROVED** [review]

File: specification/cognitiveservices/OpenAI.Inference/models/audio/common.tsp (hunk 2 above)
Comment (weidongxu-microsoft) [reply specification/cognitiveservices/OpenAI.Inference/models/audio/common.tsp:20]: This one describes the response text / string. So I'd assume we say e.g. `The Response data of content type "text/plain".`?

File: specification/cognitiveservices/OpenAI.Inference/routes/common.tsp [hunk 3]
```diff
@@ -30,3 +30,9 @@ alias MultipartFormDataRequestHeadersTraits = Azure.Core.Traits.RequestHeadersTr
   @header("content-type")
   contentType: "multipart/form-data";
 }>;
+
+alias TextPlainResponseHeadersTraits = Azure.Core.Traits.ResponseHeadersTrait<{
+  @doc("'text/plain' content type for the resonse.")
+  @header("content-type")
+  contentType: "text/plain";
```
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply specification/cognitiveservices/OpenAI.Inference/routes/common.tsp:37]: runtime, content-type header is `Content-Type:text/plain; charset=utf-8`

File: specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-09-01-preview/generated.json [hunk 4]
```diff
@@ -75,12 +82,25 @@
...
+            "$ref": "#/parameters/AudioTranscriptionOptions.language"
+          },
+          {
+            "$ref": "#/parameters/AudioTranscriptionOptions.prompt"
+          },
+          {
+            "$ref": "#/parameters/AudioTranscriptionOptions.temperature"
+          },
+          {
+            "$ref": "#/parameters/AudioTranscriptionOptions.model"
```
Comment (qiaozha) [inline specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-09-01-preview/generated.json:103]: Is it expected to spread the AudioTranscriptionOptions model here ?

Comment (weidongxu-microsoft) [reply specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-09-01-preview/generated.json:103]: I think it is expected for Swagger https://swagger.io/docs/specification/2-0/file-upload/

**Review by msyyc — APPROVED** [review]
The PR could avoid bug https://github.com/microsoft/typespec/issues/2938 so that SDK generation will not be blocked.

**Review by trrwilson — APPROVED** [review]

**Pull request merged**